giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/tokens/impl/IntrospectTokenService.java

```diff
@@ -0,0 +1,139 @@
+package edu.asu.diging.gilesecosystem.web.tokens.impl;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Date;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.nimbusds.jose.util.Base64;
+
+import edu.asu.diging.gilesecosystem.util.properties.IPropertiesManager;
+import edu.asu.diging.gilesecosystem.web.apps.IRegisteredApp;
+import edu.asu.diging.gilesecosystem.web.exceptions.AppMisconfigurationException;
+import edu.asu.diging.gilesecosystem.web.service.apps.IRegisteredAppManager;
+import edu.asu.diging.gilesecosystem.web.service.properties.Properties;
+import edu.asu.diging.gilesecosystem.web.tokens.IApiTokenContents;
+
+@Service
+public class IntrospectTokenService {
```

Comment:
now implements interface.
I'll write test cases for IntrospectTokenService and MitreidAccessTokenChecker as separate subtast under this task.